File: 8 kyu/DNA to RNA Conversion/solution.cs
namespace Converter {
  public class Converter
  {
    public string dnaToRna(string dna)
    {
      return dna.Replace('T', 'U');
    }
  }
}

_____________________________
namespace Converter {
  public class Converter
  {
    public string dnaToRna(string dna) => dna.Replace('T','U');
  }
}

_____________________________
namespace Converter {
  public class Converter
  {
    public string dnaToRna(string dna)
    {
      char[] ch = dna.ToCharArray();
      for (int i = 0; i < ch.Length; i++){
        if (ch[i] == 'T'){
          ch[i] = 'U';
        }
      }
      string rna = new string (ch);
      return rna;
    }
  }
}

_____________________________
namespace Converter {
  public class Converter
  {
    public string dnaToRna(string dna)
    {
       return dna.Length > 0? dna.Replace('T', 'U'):dna;
    }
  }
}

File: 8 kyu/DNA to RNA Conversion/solution.rs
fn dna_to_rna(dna: &str) -> String {
 dna.replace("T", "U")
}

_____________________________
fn dna_to_rna(dna: &str) -> String {
    dna.chars().map(char_conversion).collect()
}

fn char_conversion(c: char) -> char {
    if c == 'T' {
        return 'U';
    }
    
    c
}

_____________________________
fn new_char(c: char) -> char {
    return if c == 'T' { 'U' } else { c };
} 

fn dna_to_rna(dna: &str) -> String {
    let vec: Vec<char> = dna.chars().map(new_char).collect();
    let resultStr: String = vec.into_iter().collect();
    
    return resultStr;
}
      
_____________________________
fn dna_to_rna(dna: &str) -> String {
    let mut rna = String::new();
    dna.chars().for_each(|c| rna.push(if c == 'T' { 'U' } else { c }));
    rna
}
